### ISSUE TITLE
X-Traefik-Reqid header appears to be missing from backend requests since version 1.2.0 final.

### ISSUE THREAD

**Issue by Yggdrasil** [containous/traefik#1333]
### What version of Traefik are you using (`traefik version`)?
Version:      v1.2.0
Codename:     morbier
Go version:   go1.7.5
Built:        2017-03-21_10:02:52AM
OS/Arch:      linux/amd64

### What is your environment & configuration (arguments, toml...)?
Ubuntu 14.04 LTS instances on Amazon VPC. Traefik (with file-based config) in front of Varnish in front of Apache+PHP.

Our traefik.toml:
```toml
defaultEntryPoints = ["http"]
logLevel = "INFO"
traefikLogsFile = "/var/log/traefik/daemon.log"
[acme]
email = "example@example.com"
entryPoint = "https"
onDemand = true
storage = "/var/lib/traefik/acme.json"
[entryPoints.http]
address = ":80"
[entryPoints.https]
address = ":443"
[entryPoints.https.tls]
[file]
watch = false
[frontends.non-tls]
backend = "varnish"
entryPoints = ["http"]
passHostHeader = true
[frontends.tls]
backend = "varnish"
entryPoints = ["http", "https"]
passHostHeader = true
[frontends.tls.routes.test_1]
rule = "Host:www.example.com"
[backends.varnish.servers.server1]
url = "http://127.0.0.1:6081"
[web]
address = ":8081"
readonly = true
[web.statistics]
```

### What did you do?
I upgraded Traefik from 1.2.0-rc2 to 1.2.0.

### What did you expect to see?
I expect a `X-Traefik-Reqid` header to be present in the backend request. We used this header to tag and identify the request throughout our application stack, since Traefik is the entrypoint into our network.

The header was still present in 1.2.0-rc2.

### What did you see instead?
The header is missing since 1.2.0.

**Comment by emilevauge** [comment]
@Yggdrasil You are using an undocumented feature ;)
`X-Traefik-Reqid` is only set when you enable Access logs file. Could you try with this option ?

**Comment by Yggdrasil** [comment]
@emilevauge You're right. When the access log is enabled, the header returned.

I also realized now that the counter starts from 1 if Traefik is restarted, which makes it less useful for tracing a request through the stack. 

Some sort of unique ID in each request would be good though, since Traefik is often the place where requests first come into your network. That value should be unique though across restarts. Perhaps a UUID v4?

I guess this would be easier to do once header middleware (PR #1236) is in place, but we'd need a UUID generator as well. Shall I create a follow-up issue?

**Comment by Multiply** [comment]
We're also really interested in having an option to set a header such as `X-REQUEST-ID` with a randomly generated UUID, but only if it's not already set.

**Comment by ldez** [comment]
I'll close this issue, because I think the question is answered, but feel free to continue the conversation.
